app/src/main/java/com/example/meetup/Fragments/SearchFragment.java

```diff
@@ -0,0 +1,139 @@
+        posts.setAdapter(adapter);
+        LinearLayoutManager llm = new LinearLayoutManager(getContext());
+        posts.setLayoutManager(llm);
+
+        searchFind.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String searchNameQuery = searchName.getText().toString();
+                String searchCategoryQuery = searchCategory.getText().toString();
+                String searchKeyWordQuery = searchKeyWord.getText().toString();
+                String searchRoleQuery = searchRole.getText().toString();
+                queryPosts(searchNameQuery, searchCategoryQuery, searchKeyWordQuery, searchRoleQuery);
+
+                searchName.setText("");
+                searchCategory.setText("");
+                searchKeyWord.setText("");
+                searchRole.setText("");
+            }
+        });
+    }
+
+    private void queryPosts(String searchName, String searchCategory, String searchKeyWord, String searchRole) {
+        ParseQuery<Post> query = ParseQuery.getQuery(Post.class);
+        query.include(Post.KEY_USER);
+        // search for key work both in short caption and description
+        // if the key word appears in any of them, add to query result
+        if (searchKeyWord != null && searchKeyWord.length() > 0) {
+            //query.whereContains(Post.KEY_DESCRIPTION, searchKeyWord);
+            ParseQuery<Post> queryCaption = ParseQuery.getQuery(Post.class);
+            queryCaption.whereContains(Post.KEY_CAPTION, searchKeyWord);
+            ParseQuery<Post> queryDescription = ParseQuery.getQuery(Post.class);
+            queryDescription.whereContains(Post.KEY_DESCRIPTION, searchKeyWord);
+
+            List<ParseQuery<Post>> queries = new ArrayList<ParseQuery<Post>>();
+            queries.add(queryCaption);
+            queries.add(queryDescription);
+
+             query = ParseQuery.or(queries);
+             query.include(Post.KEY_USER);
+        }
+        if (searchName != null && searchName.length() > 0)
+            query.whereEqualTo(Post.KEY_STARTUP_NAME, searchName);
+        if (searchCategory != null && searchCategory.length() > 0)
+            query.whereEqualTo(Post.KEY_CATEGORY, searchCategory);
+        if (searchRole != null && searchRole.length() > 0)
+            query.whereContains(Post.KEY_ROLES, searchRole);
+
+        query.findInBackground(new FindCallback<Post>() {
+            @Override
+            public void done(List<Post> posts, ParseException e) {
+                if (e != null) {
+                    Log.e(TAG, "issue with getting posts");
+                }
+                for (Post post: posts) {
+                    Log.i(TAG, "Post " + post.getDescription() + " username " + post.getUser().getUsername());
+                }
+
+                // save received posts to list and notify adapter of new data
+                allPosts.addAll(posts);
```

Comment:
This means if user keep searching, the post list will keep growing? This seems not right behavior. And the latest search result is on the bottom of list?
This seems not correct behavior.

Reply:
i updated the code to clear the array and adapter after each search